### DISCUSSION TITLE
Git clients conflict

### DISCUSSION THREAD

**Discussion by ghost** [git-for-windows/git#5163] [General]
The repo is cloned by Git for Windows (Git patched to build with MinGW-w64?). When I tried to use the MSYS2 Git (Cygwin Git?) to do a `git pull`, I always see:

`error: Your local changes to the following files would be overwritten by merge:`

and

```
Please commit your changes or stash them before you merge.
Aborting
```

The problem is I didn't make any local changes. I can use Git for Windows to do a `git pull` without any problems.

Note: MSYS2 Git is a newer version compared to Git for Windows. I have not yet updated Git for Windows to the latest release. But I don't think this could a problem.

Reference: https://github.com/msys2/MSYS2-packages/discussions/4914

**Comment by dscho** [comment]
Does it work when you call `git status` before `git pull` (using the _same_ `git` client)?

**Reply by ghost** [reply]
I don't understand what you said. You are telling me to run `git status` with the current git client as a prepare step before I could use another git client? Or have I need to run `git status` on the other git client before I could do a `git pull` using it?

**Reply by dscho** [reply]
No, I asked to run `git status` immediately before `git pull`, with the _same_ `git`.

**Reply by ghost** [reply]
```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
  (commit or discard the untracked or modified content in submodules)

[All files in the repository will be listed as modified]

no changes added to commit (use "git add" and/or "git commit -a")
```

But after that, `git pull` worked.

**Reply by ghost** [reply]
Nope. It doesn't work. Git tells me that it updated the files. But the fact is nothing get updated. The upstream repo fixed a compilation bug. I'm complaining to them that it's not fixed at all. We both have no idea what's going on, until I open the files and find out that the old and not updated content is still there. I have to delete the whole repository and clone again, and the code now compiles just fine.

**Reply by dscho** [reply]
> after that, git pull worked.

That is indeed expected. One of the differences between MSYS2 `git.exe` and MINGW `git.exe` is that the former looks up something akin to the inode numbers that Unix has _and stores that information in the Git index_. For performance reason, this is not what Git for Windows does (even if there [is a ticket](https://github.com/git-for-windows/git/issues/3707) exploring how to improve that).

It is essentially the same challenge as trying to use the same Git index between a Windows `git.exe` and a WSL `git`: It won't work. This is an expected difference, and you will have to work around it by refreshing the index (which `git status` did for you, and which is necessarily a bit expensive to do in terms of time and I/O).

**Comment by jeremyd2019** [comment]
I suspect `core.autocrlf` - line ending differences being the 'changes' it's talking about.

**Comment by tboegi** [comment]
Can you run a 
git ls-files --eol
on both systems ?

**Reply by ghost** [reply]
@jeremyd2019 was right. Git for Windows uses CRLF. MSYS2 Git uses LF.
